google/cloud/spanner_dbapi/connection.py

```diff
@@ -160,8 +199,13 @@ def commit(self):
         if self._autocommit:
             warnings.warn(AUTOCOMMIT_MODE_WARNING, UserWarning, stacklevel=2)
         elif self._transaction:
-            self._transaction.commit()
-            self._release_session()
+            try:
+                self._transaction.commit()
+                self._release_session()
+                self._statements = []
```

Comment:
Better to keep these out of the try block if you're only catching errors for the first line.

Reply:
As I'm calling `self.commit()` on exception, these lines will be executed twice, if we'll move them outside the `try` block